Ubuntu: https://bugs.launchpad.net/bugs/106995
Upstream: http://bugzilla.gnome.org/show_bug.cgi?id=518405
Description: Handle scrolling differently when using alternate screen or
 scrolling is restricted.

Index: deepinvte-0.28.0/python/deepinvte.defs
===================================================================
--- deepinvte-0.28.0.orig/python/deepinvte.defs	2009-04-17 18:51:43.000000000 +0200
+++ deepinvte-0.28.0/python/deepinvte.defs	2011-05-17 12:18:16.656757671 +0200
@@ -202,6 +202,15 @@
   (return-type "gboolean")
 )
 
+(define-method set_alternate_screen_scroll
+  (of-object "DeepinvteTerminal")
+  (c-name "deepinvte_terminal_set_alternate_screen_scroll")
+  (return-type "none")
+  (parameters
+    '("gboolean" "scroll")
+  )
+)
+
 (define-method set_scroll_background
   (of-object "DeepinvteTerminal")
   (c-name "deepinvte_terminal_set_scroll_background")
Index: deepinvte-0.28.0/src/deepinvte.c
===================================================================
--- deepinvte-0.28.0.orig/src/deepinvte.c	2011-02-16 20:42:09.000000000 +0100
+++ deepinvte-0.28.0/src/deepinvte.c	2011-05-17 12:18:16.676757671 +0200
@@ -11409,8 +11409,9 @@
 		return FALSE;
 	}
 
-	if (terminal->pvt->screen == &terminal->pvt->alternate_screen ||
-		terminal->pvt->normal_screen.scrolling_restricted) {
+	if (terminal->pvt->alternate_screen_scroll &&
+		(terminal->pvt->screen == &terminal->pvt->alternate_screen ||
+		terminal->pvt->normal_screen.scrolling_restricted)) {
 		char *normal;
 		gssize normal_length;
 		const gchar *special;
@@ -13063,6 +13064,22 @@
         g_object_notify (G_OBJECT (terminal), "scroll-on-keystroke");
 }
 
+/**
+ * deepinvte_terminal_set_alternate_screen_scroll:
+ * @terminal: a #DeepinvteTerminal
+ * @scroll: %TRUE if the terminal should send keystrokes for scrolling when using alternate screen
+ *
+ * Controls whether or not the terminal will send keystrokes for scrolling
+ * when using alternate screen or scrolling is restricted.
+ *
+ */
+void
+deepinvte_terminal_set_alternate_screen_scroll(DeepinvteTerminal *terminal, gboolean scroll)
+{
+	g_return_if_fail(DEEPINVTE_IS_TERMINAL(terminal));
+	terminal->pvt->alternate_screen_scroll = scroll;
+}
+
 static void
 deepinvte_terminal_real_copy_clipboard(DeepinvteTerminal *terminal)
 {
Index: deepinvte-0.28.0/src/deepinvte.h
===================================================================
--- deepinvte-0.28.0.orig/src/deepinvte.h	2011-02-16 20:27:16.000000000 +0100
+++ deepinvte-0.28.0/src/deepinvte.h	2011-05-17 12:18:16.676757671 +0200
@@ -310,6 +310,8 @@
 void deepinvte_terminal_set_scroll_on_output(DeepinvteTerminal *terminal, gboolean scroll);
 void deepinvte_terminal_set_scroll_on_keystroke(DeepinvteTerminal *terminal,
 					  gboolean scroll);
+void deepinvte_terminal_set_alternate_screen_scroll(DeepinvteTerminal *terminal,
+					      gboolean scroll);
 
 /* Set the color scheme. */
 void deepinvte_terminal_set_color_dim(DeepinvteTerminal *terminal,
Index: deepinvte-0.28.0/src/deepinvte-private.h
===================================================================
--- deepinvte-0.28.0.orig/src/deepinvte-private.h	2011-02-03 20:29:32.000000000 +0100
+++ deepinvte-0.28.0/src/deepinvte-private.h	2011-05-17 12:18:16.676757671 +0200
@@ -280,6 +280,7 @@
 	gboolean scroll_on_output;
 	gboolean scroll_on_keystroke;
 	long scrollback_lines;
+	gboolean alternate_screen_scroll;
 
 	/* Cursor shape */
 	DeepinvteTerminalCursorShape cursor_shape;
